### ISSUE TITLE
H125: purge overlapping six-bar training labels at walk-forward boundaries

### ISSUE THREAD

**Issue by yuyank-code** [yuyank-code/bitcoin-ml-trading#73]
## Critical leakage finding — 2026-09-15

Audit of `trading_bot.py::walk_forward()` found that the expanding training set ends exactly at the outer test start:

```python
train = out.iloc[:start]
test = out.iloc[start:end]
```

But `future_return` is a six-bar forward label:

```python
d.Close.shift(-cfg.horizon_bars) / d.Close - 1
```

Therefore the final `horizon_bars` training observations immediately before `start` have labels whose outcome windows extend into the test period. For `horizon_bars=6`, training rows `start-6 ... start-1` use future closes at `start ... start+5`. The model is consequently trained on labels containing prices from the period it is supposed to predict.

### Why this is blocking
This is direct label-overlap leakage at every walk-forward boundary. The predictor can be strictly evaluated on OOS rows while still having trained on outcomes that occur inside the OOS interval. Any OOS performance from this walk-forward path is contaminated until the overlap is purged and results are regenerated.

### Required remediation
1. Define each sample's label interval explicitly: `[signal_time, label_end_time]`.
2. Before fitting fold `start`, purge every training row whose label end is >= the first test decision time.
3. For the fixed six-bar horizon, the minimum expanding-window purge is six bars; implement from timestamps/label intervals rather than a magic constant where possible.
4. Add an assertion that `max(train.label_end_time) < min(test.signal_time)` for every fold.
5. Add a fixture proving that the last six pre-boundary labels are excluded while earlier labels remain eligible.
6. Keep the existing event-time feature alignment, nested policy selection, cost/slippage, and H106 multiplicity accounting unchanged.
7. Regenerate all affected OOS prediction vectors and economic ledgers; pre-fix results are not admissible evidence.
8. Add temporal non-interference: changing prices inside the future test interval must not change the fitted training labels/predictions for the preceding fold.

### Testable hypothesis
After purging, no training label's information interval overlaps the first OOS decision interval. Changing any future OOS price after the fold boundary cannot change the training dataset or the resulting earlier prediction.

### Literature motivation
Purging overlapping labels is a core financial-ML validation requirement. Recent 2026 validation implementations explicitly remove training samples whose label windows overlap test windows and separately apply embargo where serial dependence warrants it. The project's six-bar overlapping target makes this boundary condition unavoidable.

### Promotion rule
No candidate trained through this walk-forward implementation may be promoted, and no pre-fix OOS metric may be cited as clean evidence, until H125 passes and affected experiments are regenerated.
